src/main/java/ch/njol/skript/lang/SkriptParser.java

```diff
@@ -213,40 +213,70 @@ public static <T extends SyntaxElement> T parseStatic(String expr, Iterator<? ex
 		}
 	}
 
+	/**
+	 * Attempts parsing into one of the given syntax element classes.
+	 */
 	@Nullable
 	private <T extends SyntaxElement> T parse(Iterator<? extends SyntaxElementInfo<? extends T>> source) {
+		ParsingStack parsingStack = getParser().getParsingStack();
+
 		ParseLogHandler log = SkriptLogger.startParseLogHandler();
 		try {
+			// Iterate over all SyntaxElementInfos
 			while (source.hasNext()) {
 				SyntaxElementInfo<? extends T> info = source.next();
+
+				// Iterate over its patterns
 				patternsLoop: for (int i = 0; i < info.patterns.length; i++) {
 					log.clear();
 					try {
 						String pattern = info.patterns[i];
 						assert pattern != null;
+
 						ParseResult res;
 						try {
+							// Push the element onto the parsing stack
+							parsingStack.push(new ParsingStack.Element(info, i));
+
 							res = parse_i(pattern, 0, 0);
 						} catch (MalformedPatternException e) {
+							// Malformed pattern, add more context and propagate exception upwards
 							String message = "pattern compiling exception, element class: " + info.c.getName();
+
 							try {
 								JavaPlugin providingPlugin = JavaPlugin.getProvidingPlugin(info.c);
 								message += " (provided by " + providingPlugin.getName() + ")";
-							} catch (IllegalArgumentException | IllegalStateException ignored) {}
+							} catch (IllegalArgumentException | IllegalStateException ignored) { }
```

Comment:
Wouldn't it also be beneficial to add these exceptions to the parse stack exception list instead of ignoring? It may be very low chances or not possible, but it might if something changes.

Reply:
These aren't related, they only occur in the case that the class does not come from a plugin. I don't think an exception telling us that the SE class isn't from a plugin (or not as far as the server knows) is helpful.